.github/workflows/govet.yml

```diff
@@ -0,0 +1,19 @@
+name: govet
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'doc/**'
+      - 'enhancements/**'
+      - '*.md'
+
+jobs:
+  govet_test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+      - name: Run go vet
+        run: go vet ./...
```

Comment:
We have go vet in our test execution `--vet=all`.
Will this check something different?

Reply:
Nope it's just for visibility reason, we have dedicated gofmt action as well. So I think it made sense to have a separate action for govet to avoid cluttered output. govet is also implemented in golangci lint, but it also has the same troubles.

Reply:
We can have it separated, but we should have it in only one place.

Reply:
Yes that makes more sense, I guess. I will remove them from golangci lint and test args for vet. Also, I will do the same for fmt as well as it would be redundant to run the test twice.

Reply:
@sbdtu5498 it is ready to merge, can you address this comment? thanks

Reply:
@rootfs I will do it by today positively.

Reply:
@sbdtu5498 sounds good

Reply:
@SamYuan1990 @sbdtu5498 ^^

Reply:
Just a sec. I will do it right now so that you can merge this

Reply:
Okay sure. Sound's good to me. I am adding new PR on top of this one. Kindly merge so that I can fetch it.